Licenses redesign

Stevic-Nenad opened the pull request:
## Resolves: #333 

- Moved licenses to a new page only reachable through `/about`
- Grouped licenses by publisher, or if none available, by the package name split by `/`. For example, all licenses starting with `@types/` are now grouped together for better readability.
- Adjusted existing tests so they don't break
- Added a new test suite for licenses only.

Philippluca reviewed (changes requested):
REQ: Navigation über Link und Back setzt Sprache auf EN / Default zurück. Andere Navigation (z.B. im Footer) haben diesen Effekt nicht.

Stevic-Nenad commented:
- Refactored Link behaviour on the about page when redirecting to the licenses page. Using react router doms link instead to not reload page when clicking the link.
- Refactored Dockerfile so npm install happens earlier on and is cached, as discussed with @Philippluca. It's one of the more compute intensive operations in the image creation process, so we only want to run it when necessary.
- Transitions on the accordions should be nice an quick now

Philippluca reviewed (approved):
LGTM
